src/base/RegexPattern.cc

```diff
@@ -8,34 +8,29 @@
 
 #include "squid.h"
 #include "base/RegexPattern.h"
-#include "base/TextException.h"
 #include "debug/Stream.h"
 #include "sbuf/Stream.h"
 
 #include <iostream>
-#include <utility>
 
-RegexPattern::RegexPattern(const SBuf &aPattern, const int aFlags):
+RegexPattern::RegexPattern(const SBuf &aPattern, const std::regex::flag_type aFlags):
     pattern(aPattern),
-    flags(aFlags)
+    regex(aPattern.rawContent(), aPattern.length(), aFlags)
 {
-    memset(&regex, 0, sizeof(regex)); // paranoid; POSIX does not require this
-    if (const auto errCode = regcomp(&regex, pattern.c_str(), flags)) {
-        char errBuf[256];
-        // for simplicity, ignore any error message truncation
-        (void)regerror(errCode, &regex, errBuf, sizeof(errBuf));
-        // POSIX examples show no regfree(&regex) after a regcomp() error;
-        // presumably, regcom() frees any allocated memory on failures
-        throw TextException(ToSBuf("POSIX regcomp(3) failure: (", errCode, ") ", errBuf,
-                                   Debug::Extra, "regular expression: ", pattern), Here());
-    }
-
+    assert(aFlags != 0);
+    assert((aFlags & (std::regex::basic|std::regex::extended)) != 0);
```

Comment:
We should explain this puzzling assertion. I do not insist on any specific wording, but here is a sketch:

```suggestion
    // this class supports other syntax variations, but its current users must
    // support one of these two for backward compatibility reasons, and we check
    // that they have not forgotten to do so
    assert((aFlags & (std::regex::basic|std::regex::extended)) != 0);
```